sklearn/naive_bayes.py

```diff
@@ -593,6 +595,16 @@ def _update_class_log_prior(self, class_prior=None):
             self.class_log_prior_ = np.full(n_classes, -np.log(n_classes))
 
     def _check_alpha(self):
+        # TODO(1.4): Replace w/ deprecation of self.force_alpha
+        # See gh #22269
+        _force_alpha = self.force_alpha
+        if _force_alpha == "warn":
+            _force_alpha = False
```

Comment:
Crazy idea: What do you think about not warning when `alpha == 1.0` or even `alpha >= _ALPHA_MIN`?

Reply:
I prefer warning even in those cases, if nothing else than to just communicate to the user the otherwise difficult-to-see change in behavior. Even if it doesn't affect their use _right now_, it may in the future.